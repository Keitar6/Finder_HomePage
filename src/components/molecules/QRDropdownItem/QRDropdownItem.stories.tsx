import { Meta, StoryObj } from '@storybook/react';
import GlobalStyle from '../../../theme/globalStyles';
import ThemeProvider from '../../../theme/ThemeContext';
import QRDropdownItem from './QRDropdownItem';

const meta = {
  title: 'Molecules/QRDropdownItem',
  component: QRDropdownItem,
  tags: ['autodocs'],
  args: { placeholder: 'Search flats...' },
  argTypes: {

  },
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
  parameters: {
    backgrounds: { default: 'light' },
  },
} satisfies Meta<typeof QRDropdownItem>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {};
