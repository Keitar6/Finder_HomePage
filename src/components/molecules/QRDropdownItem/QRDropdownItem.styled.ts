import styled from 'styled-components';
import initialTheme from '../../../theme/initialTheme';

export const Wrapper = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  padding-top: 4rem;

  min-width: 14rem;
`;
export const QRDropdown = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  gap: 1rem;
  padding: 26px 8px;
  border: 1px solid ${initialTheme.palette.inputGrey};
  border-radius: 8px;
  background-color: ${initialTheme.palette.light};

  min-width: 14rem;
  position: absolute;
`;

export const QRImage = styled.img`
  width: 10rem;
`;
export const Texts = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;

  gap: 0.3rem;
`;
