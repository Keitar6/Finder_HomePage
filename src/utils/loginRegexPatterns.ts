import { InputVariant, RegexPat } from '../components/atoms/Input/Input.types';
import { Getters } from '../types/shared.types';

type LoginRegexProps = Getters<InputVariant, RegexPat>;

const loginRegexPattern: LoginRegexProps = {
  search: {
    pattern: /^[a-zA-Z0-9]+$/,
    required: true,
    maxLength: 36,
    minLength: 8,
  },
};

export default loginRegexPattern;
