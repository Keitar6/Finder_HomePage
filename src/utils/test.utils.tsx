import { PropsWithChildren } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import ThemeProvider from '../theme/ThemeContext';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  initialUrl?: string;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    initialUrl = '/',
    ...renderOptions
  }: ExtendedRenderOptions = {},
) {
  const Wrapper = ({ children }: PropsWithChildren<object>): JSX.Element => (
    <ThemeProvider>
        {children}
    </ThemeProvider>
  );

  return { ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export * from '@testing-library/react';
