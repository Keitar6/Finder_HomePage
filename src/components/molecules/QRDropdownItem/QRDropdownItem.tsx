import Typography from '../../atoms/Typography/Typography';
import Button from '../../atoms/Button/Button';
import QRCode from '../../../assets/img/QRCode.svg';
import CenterWrapper from '../../atoms/CenterWrapper/CenterWrapper';
import * as Styled from './QRDropdownItem.styled';

const QRDropdownItem = () => (
  <Styled.Wrapper>
    <Styled.QRDropdown>
      <Styled.QRImage src={QRCode} alt='QR Code --> Download application' />
      <Styled.Texts>
        <Typography variant='smallParagraph' style={{ fontWeight: 'bold', textAlign: 'center' }}>
          Free Credit Score
        </Typography>
        <Typography variant='smallParagraph' style={{ textAlign: 'center' }}>
          Scan to download the ultimate money app
        </Typography>
      </Styled.Texts>
      <CenterWrapper>
        <Button variant='hoverReverse' style={{ borderRadius: '100rem' }}>
          Get it now
        </Button>
      </CenterWrapper>
    </Styled.QRDropdown>
  </Styled.Wrapper>
);

export default QRDropdownItem;
