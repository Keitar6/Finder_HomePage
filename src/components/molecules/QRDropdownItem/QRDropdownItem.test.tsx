import { cleanup } from '@testing-library/react';
import { renderWithProviders } from '../../../utils/test.utils';
import QRDropdownItem from './QRDropdownItem';

describe('QRDropdownItem', () => {
  afterEach(cleanup);
  describe('QRDropdownItem component', () => {

    const QRDropdownItemRender: any = () => renderWithProviders(<QRDropdownItem />, {});
    it('should render correctly', () => {
      const QRDropdownItem = QRDropdownItemRender();
      expect(QRDropdownItem).toMatchSnapshot();
    });
  });
});
