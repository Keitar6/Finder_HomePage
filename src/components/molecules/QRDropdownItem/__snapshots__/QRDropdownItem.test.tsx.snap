// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QRDropdownItem QRDropdownItem component should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-iGgWBj eFYmRU"
      >
        <div
          class="sc-gsFSXq gnYjrc"
        >
          <img
            alt="QR Code --> Download application"
            class="sc-kAyceB kjPvUD"
            src="QRCode.svg"
          />
          <div
            class="sc-imWYAI qAOLQ"
          >
            <p
              style="font-family: Roboto, Arial, sans-serif; font-weight: bold; font-size: 14px; color: rgb(36, 42, 47); text-align: center;"
            >
              Free Credit Score
            </p>
            <p
              style="font-family: Roboto, Arial, sans-serif; font-weight: 500; font-size: 14px; color: rgb(36, 42, 47); text-align: center;"
            >
              Scan to download the ultimate money app
            </p>
          </div>
          <div
            class="sc-dcJsrY AyQQA"
          >
            <button
              class="sc-gEvEer sc-fqkvVR caaaGq dFiuBA"
              style="border-radius: 100rem;"
            >
              Get it now
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-iGgWBj eFYmRU"
    >
      <div
        class="sc-gsFSXq gnYjrc"
      >
        <img
          alt="QR Code --> Download application"
          class="sc-kAyceB kjPvUD"
          src="QRCode.svg"
        />
        <div
          class="sc-imWYAI qAOLQ"
        >
          <p
            style="font-family: Roboto, Arial, sans-serif; font-weight: bold; font-size: 14px; color: rgb(36, 42, 47); text-align: center;"
          >
            Free Credit Score
          </p>
          <p
            style="font-family: Roboto, Arial, sans-serif; font-weight: 500; font-size: 14px; color: rgb(36, 42, 47); text-align: center;"
          >
            Scan to download the ultimate money app
          </p>
        </div>
        <div
          class="sc-dcJsrY AyQQA"
        >
          <button
            class="sc-gEvEer sc-fqkvVR caaaGq dFiuBA"
            style="border-radius: 100rem;"
          >
            Get it now
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
